versus/js/05-asynchronous-computation/os-events-description.html

```diff
@@ -0,0 +1,59 @@
+<h4>What This Code Does</h4>
+<p>
+	It is common for a program to instruct the operating system to perform <em>long-running computations</em> or
+	to respond to <em>events from external sources</em>. A long running computation could be copying a large file or
+	scheduling a timer. The external events can be a GUI program user inputs, or for a server program activity in the network for example.
+	In this role the operating system (OS) is a service provider which exposes blocking and non-blocking methods.
+	Delegating thread management and polling loops to the OS by using non-blocking asynchronous OS methods,
+	the C++ program is reduced in its complexity and is made less error-prone without being unresponsive during
+	the execution time of background computations.
+</p>
+<p>
+	Asynchrony can also be achieved with <em>long-running internal computations</em> by moving them into background threads.
+	To avoid blocking the main thread of execution the program instructs the OS to execute the background thread and signal
+	the main thread when the background thread has exited.
+</p>
+<p>
+	Calling non-blocking OS methods from C++ is trivial (it's what C++ is made for), but to get notified of external events, once an
+	event loop must be implemented and registered with the OS. Also OS-dependent C-callbacks or background threads
+	have to be registered for each non-blocking method.
+</p>
+<p>
+	<a href="https://www.boost.org/doc/libs/1_68_0/doc/html/boost_asio.html">Boost.Asio</a>
+	provides a platform-independent asynchronous api for common long-running tasks in networking, timers, serial ports and signals.
```

Comment:
The discussion on Boost and the previous paragraph should really be moved into one of the next two sections (what's the same/what's different). Let's keep the top on focus to describe the problem in general terms and not in specific terms to C++.